src/components/Table/TableConfirmButtonsRow.js

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { createPortal } from 'react-dom';
+import { ConfirmButton, CancelButton } from '../InlineEdit';
+import { noop } from '../../common/helpers';
+
+class TableConfirmButtonsRow extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+
+    this.saveRowDimensions = this.saveRowDimensions.bind(this);
+    this.handleScroll = this.handleScroll.bind(this);
+    this.handleResize = this.handleResize.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchClientDimensions();
+    window.addEventListener('scroll', this.handleScroll);
+    window.addEventListener('resize', this.handleResize);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleScroll);
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  saveRowDimensions(element) {
+    if (element) {
+      this.element = element;
+    }
+    if (this.element) {
+      this.setState({
+        rowDimensions: this.element.getBoundingClientRect()
+      });
+    }
+  }
+
+  handleScroll(event) {
+    this.saveRowDimensions();
+  }
+
+  handleResize(event) {
+    this.fetchClientDimensions();
+    this.saveRowDimensions();
+  }
+
+  fetchClientDimensions() {
+    this.setState({
+      window: {
+        width: document.documentElement.clientWidth,
+        height: document.documentElement.clientHeight
+      }
+    });
+  }
+
+  renderConfirmButtons() {
+    const divStyle = this.state.rowDimensions
+      ? this.props.buttonsPosition(this.state.window, this.state.rowDimensions)
+      : {};
+
+    const buttonsClass = this.props.buttonsClass('inline-edit-buttons');
+    return (
+      <div style={divStyle} className={buttonsClass} key="confirmButtons">
+        <ConfirmButton
+          bsStyle="primary"
+          key="confirm"
+          aria-label={this.props.messages.confirmButtonLabel}
+          onMouseUp={() => this.props.onConfirm()}
+        />
+        <CancelButton
+          bsStyle="default"
+          key="cancel"
+          aria-label={this.props.messages.cancelButtonLabel}
+          onMouseUp={() => this.props.onCancel()}
+        />
+      </div>
+    );
+  }
+
+  render() {
+    const editing = this.props.isEditing();
+    const rowClass = editing ? 'editing' : '';
+
+    const elements = [
+      <tr ref={this.saveRowDimensions} className={rowClass} key="tableRow">
+        {this.props.children}
+      </tr>
+    ];
+
+    if (editing && (this.element || this.props.buttonsMountpoint)) {
+      elements.push(
+        // mount the confirm buttons below the table
+        createPortal(
+          this.renderConfirmButtons(),
+          this.props.buttonsMountpoint ||
+            this.element.closest('table').parentNode
+        )
+      );
+    }
+
+    return elements;
+  }
+}
+
+TableConfirmButtonsRow.shouldComponentUpdate = true;
```

Comment:
Is there a need for this?

Reply:
There is. See https://github.com/reactabular/reactabular/blob/master/docs/table/overriding-should-component-update.md and the reactabular code.
The default imlementation of `shouldComponentUpdate` allows updates only when columns or row data have changed. This is too strict in our case. We need to re-render components when the editing state is switched independently of row data change.